tests/extmod_hardware/machine_uart_irq_rx.py

```diff
@@ -0,0 +1,78 @@
+# Test machine.UART.IRQ_RX firing for each character received.
+#
+# IMPORTANT: This test requires hardware connections: the UART TX and RX
+# pins must be wired together.
+
+try:
+    from machine import UART
+
+    UART.IRQ_RX
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+import time, sys
+
+byte_by_byte = False
+# Configure pins based on the target.
+if "esp32" in sys.platform:
+    uart_id = 1
+    tx_pin = 4
+    rx_pin = 5
+elif "pyboard" in sys.platform:
+    uart_id = 4
+    tx_pin = None  # PA0
+    rx_pin = None  # PA1
+elif "samd" in sys.platform and "ItsyBitsy M0" in sys.implementation._machine:
+    uart_id = 0
+    tx_pin = "D1"
+    rx_pin = "D0"
+    byte_by_byte = True
+elif "samd" in sys.platform and "ItsyBitsy M4" in sys.implementation._machine:
+    uart_id = 3
+    tx_pin = "D1"
+    rx_pin = "D0"
+elif "samd" in sys.platform and "Seeed" in sys.implementation._machine:
+    uart_id = 0
+    tx_pin = "A4_D4"
+    rx_pin = "A5_D5"
+elif "nrf" in sys.platform:
+    uart_id = 0
+    tx_pin = None
+    rx_pin = None
+elif "renesas-ra" in sys.platform:
+    uart_id = 9
+    tx_pin = None  # P602 @ RA6M2
+    rx_pin = None  # P601 @ RA6M2
+else:
+    print("SKIP")
```

Comment:
I suggest keeping this line as `print("Please add support for this test on this platform.")`, because the test should not really be skipped on boards that do support this feature.

Reply:
Done. But we know it will fail for the CC3200, since it supports IRQ_RX but cannot pass the test.

Reply:
OK, for cc3200 it can have an explicit section here that skips that port.  And a comment.  Eg:
```py
elif sys.platform == "cc3200":
    # CC3200 doesn't work because it's too slow and has an allocation error in the handler.
    print("SKIP")
    raise SystemExit
```

Reply:
Done.